cmd/promtool/backfill_test.go

```diff
@@ -0,0 +1,265 @@
+	require.NoError(t, err)
+	series := queryblock(t, q, labels.MustNewMatcher(labels.MatchRegexp, "", ".*"))
+	require.Equal(t, expectedSeries, series)
+}
+func TestBackfill(t *testing.T) {
+	tests := []struct {
+		ToParse  string
+		IsOk     bool
+		Expected struct {
+			MinTime int64
+			MaxTime int64
+			Series  map[string][]tsdbutil.Sample
+		}
+	}{
+		{
+			ToParse: `# EOF`,
+			IsOk:    true,
+		},
+		{
+			ToParse: `# HELP http_requests_total The total number of HTTP requests.
+# TYPE http_requests_total counter
+http_requests_total{code="200"} 1021 1565133713989
+http_requests_total{code="400"} 1 1565133713990
+# EOF
+`,
+			IsOk: true,
+			Expected: struct {
+				MinTime int64
+				MaxTime int64
+				Series  map[string][]tsdbutil.Sample
+			}{
+				MinTime: 1565133713989000,
+				MaxTime: 1565133713990000,
+				Series:  map[string][]tsdbutil.Sample{"{__name__=\"http_requests_total\", code=\"200\"}": []tsdbutil.Sample{backfillSample{t: 1565133713989000, v: 1021}}, "{__name__=\"http_requests_total\", code=\"400\"}": []tsdbutil.Sample{backfillSample{t: 1565133713990000, v: 1}}},
+			},
+		},
+		{
+			ToParse: `# HELP http_requests_total The total number of HTTP requests.
+# TYPE http_requests_total counter
+http_requests_total{code="200"} 1022 1565133713989
+http_requests_total{code="400"} 2 1565133713990
+# EOF
+`,
+			IsOk: true,
+			Expected: struct {
+				MinTime int64
+				MaxTime int64
+				Series  map[string][]tsdbutil.Sample
+			}{
+				MinTime: 1565133713989000,
+				MaxTime: 1565133713990000,
+				Series:  map[string][]tsdbutil.Sample{"{__name__=\"http_requests_total\", code=\"200\"}": []tsdbutil.Sample{backfillSample{t: 1565133713989000, v: 1022}}, "{__name__=\"http_requests_total\", code=\"400\"}": []tsdbutil.Sample{backfillSample{t: 1565133713990000, v: 2}}},
+			},
+		},
+		{
+			ToParse: `# HELP http_requests_total The total number of HTTP requests.
+# TYPE http_requests_total counter
+http_requests_total{code="200"} 1023 1395066363000
+http_requests_total{code="400"} 3 1395066363000
+# EOF
+`,
+			IsOk: true,
+			Expected: struct {
+				MinTime int64
+				MaxTime int64
+				Series  map[string][]tsdbutil.Sample
+			}{
+				MinTime: 1395066363000000,
+				MaxTime: 1395066363000000,
+				Series:  map[string][]tsdbutil.Sample{"{__name__=\"http_requests_total\", code=\"200\"}": []tsdbutil.Sample{backfillSample{t: 1395066363000000, v: 1023}}, "{__name__=\"http_requests_total\", code=\"400\"}": []tsdbutil.Sample{backfillSample{t: 1395066363000000, v: 3}}},
+			},
+		},
+		{
+			ToParse: `# HELP something_weird Something weird
+		# TYPE something_weird gauge
+		something_weird{problem="infinite timestamp"} +Inf -3982045
+		# EOF
+		`,
+			IsOk: false,
+		},
+		{
+			ToParse: `# HELP rpc_duration_seconds A summary of the RPC duration in seconds.
+		# TYPE rpc_duration_seconds summary
+		rpc_duration_seconds{quantile="0.01"} 3102
+		rpc_duration_seconds{quantile="0.05"} 3272
```

Comment:
I guess this fails because the timestamps are missing. Is that really what you wanted to test?